workspaces/local-cli/src/shared/verify/verify.ts

```diff
@@ -108,16 +107,45 @@ export async function verifyTask(
   })();
 
   cli.log(
-    '\n' + fromOptic(colors.underline(`Testing task ${colors.bold(taskName)}`))
+    '\n' + fromOptic(colors.underline(`Checking task ${colors.bold(taskName)}`))
   );
 
   if (mode === null) {
-    return cli.log(fromOptic(colors.red(`Invalid task configuration. `)));
+    cli.log(fromOptic(colors.red(`Invalid task configuration. `)));
+    return false;
   }
 
   let passedAll = false;
 
+  if (mode === Modes.Test) {
+    if (dependent) {
+      passedAll = false;
+      cli.log(
+        fromOptic(
+          colors.red(
+            `Test tasks can not depend on other test tasks. ${colors.bold(
+              foundTask!.useTask!
+            )} cannot be used here.`
+          )
+        )
+      );
+    } else {
+      cli.log(
+        fromOptic(
+          colors.green(
+            `This is a test task, checking dependent task ${colors.bold(
+              foundTask!.useTask!
+            )}`
+          )
+        )
+      );
+
+      passedAll = await verifyTask(cli, foundTask!.useTask!, true);
+    }
+  }
+
   if (mode == Modes.Recommended) {
+    const startConfig = await TaskToStartConfig(foundTask!);
```

Comment:
the ! all over the place makes me think that the types aren't right, unless you're running into some ts limitation